framework/src/controller/application_state.js

```diff
@@ -81,33 +79,18 @@ class ApplicationState {
 	/**
 	 * Updates broadhash and height values.
 	 *
+	 * @param {broadhash, height} parameters - broadhash and height to update
+	 *
 	 * @returns {Promise.<boolean, Error>}
 	 */
-	async update(blocks) {
-		if (!blocks) {
-			throw new TypeError('Argument blocks should be an array.');
-		}
+	async update({ broadhash, height }) {
```

Comment:
In this scope the `broadhash` and `height` both are required and coupled to each other. So we we add some assertion here that user must pass both to update the system state. Otherwise we throw some error.